### PR TITLE
0.6.7

### PR DESCRIPTION
**Pull request by oetiker**
SECURITY when sending multiple mails in the same connection
the recipient list was never cleared. This caused mails to go
to all the recipients of mails sent previously in the same connection.

### CODE REVIEW

File: t/end-to-end.t [hunk 1]
```diff
@@ -490,3 +490,55 @@ sub loginAuth {
...
+
+        quit     => 1,
+        sub {
+            my ($smtp, $resp) = @_;
+            ok(!($resp->error), 'Allowed mail accepted by proxy using login auth');
+
+            my @calls = @{$testApi->calledWith};
+            is scalar(@calls), 2, 'Made two calls to the API';
+            is_deeply $calls[0]->{to}, ['r1@foobaz.com'],
+                'Correrct Recipients for first mail';
```
Comment (oetiker) [inline t/end-to-end.t:538]:
```suggestion
                'Correct Recipients for first mail';
```

File: t/end-to-end.t [hunk 2]
```diff
@@ -490,3 +490,55 @@ sub loginAuth {
...
+            is_deeply $calls[1]->{to}, ['r2@foobaz.com','r3@gugus.li'],
+                'Correrct Recipients for second mail';
```
Comment (oetiker) [inline t/end-to-end.t:540]:
```suggestion
                'Correct Recipients for second mail';
```

File: t/end-to-end.t [hunk 3]
```diff
@@ -490,3 +490,55 @@ sub loginAuth {
...
+            $done->();
+        }
+    );
+}
```
Comment (oetiker) [inline t/end-to-end.t:544]:
```suggestion
}
```

File: lib/SMTPProxy.pm [hunk 4]
```diff
@@ -50,6 +50,12 @@ sub setup {
         });
         $connection->mail(sub {
             my ($from, $parameters) = @_;
+            # reset the collected data except for the.
```
Comment (zaucker) [inline lib/SMTPProxy.pm:53]: Something missing in the comment

**Review by zaucker — APPROVED** [review]
Please fix comment

File: lib/SMTPProxy.pm (hunk 4 above)
Comment (oetiker) [reply lib/SMTPProxy.pm]: danke fixed

**Pull request merged**